adminSiteClient/SiteRedirectsIndexPage.tsx

```diff
@@ -0,0 +1,184 @@
+                    {redirect.target}
+                </a>
+            </td>
+            <td>
+                <button
+                    className="btn btn-danger"
+                    onClick={() => onDelete(redirect)}
+                >
+                    Delete
+                </button>
+            </td>
+        </tr>
+    )
+}
+
+type FormData = {
+    source: string
+    target: string
+}
+
+export default function SiteRedirectsIndexPage() {
+    const { admin } = useContext(AdminAppContext)
+    const [redirects, setRedirects] = useState<Redirect[]>([])
+    const [error, setError] = useState<string | null>(null)
+    const {
+        register,
+        formState: { errors, isSubmitting },
+        handleSubmit,
+        watch,
+    } = useForm<FormData>()
+    const source = watch("source")
+
+    useEffect(() => {
+        admin
+            .getJSON("/api/site-redirects.json")
+            .then((json) => {
+                setRedirects(json.redirects)
+            })
+            .catch(() => {
+                setError("Error loading redirects.")
+            })
+    }, [admin])
+
+    function validateTarget(value: string) {
+        if (value === source) {
+            return "Source and target cannot be the same."
+        }
+        return undefined
+    }
+
+    async function onSubmit(data: FormData) {
+        setError(null)
+        const json = await admin.requestJSON(
+            "/api/site-redirects/new",
+            data,
+            "POST"
+        )
+        if (json.success) {
+            setRedirects([json.redirect, ...redirects])
+        } else {
+            setError("Error creating a redirect.")
+        }
+    }
+
+    async function handleDelete(redirect: Redirect) {
+        if (!window.confirm(`Delete the redirect from ${redirect.source}?`))
+            return
+        setError(null)
+        const json = await admin.requestJSON(
+            `/api/site-redirects/${redirect.id}`,
+            {},
+            "DELETE"
+        )
+        if (json.success) {
+            setRedirects(redirects.filter((r) => r.id !== redirect.id))
+        } else {
+            setError("Error deleting the redirect.")
+        }
+    }
+
+    return (
+        <AdminLayout title="Site Redirects">
+            <main>
+                <form onSubmit={handleSubmit(onSubmit)}>
+                    <div className="form-row">
+                        <div className="form-group col-12 col-md-4">
+                            <label>Source</label>
+                            <input
+                                className={cx("form-control", {
+                                    "is-invalid": errors.source,
+                                })}
+                                {...register("source", {
+                                    required: "Please provide a source.",
+                                    pattern: {
+                                        value: VALID_PATTERN,
```

Comment:
In addition to checking for a leading slash I think we should also either disallow query params/fragments via the regex or strip them out before sending the redirect to the API.